### PR TITLE
fix: TabView does not render item content with bindings

### LINKED ISSUE #4875: Binding to TabView.TabItemsSource is not working
## Current behavior

When binding `TabItemsSource` to an `ObservableCollection`, tabs are not rendered.

## Expected behavior

TabItemsSource can be bound via x:Bind

## How to reproduce it (as minimally and precisely as possible)

See https://github.com/unoplatform/uno/issues/4628

## Environment

Nuget Package:

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### LINKED ISSUE #4895: TabView does not correctly render TabViewItem content with bindings
## Current behavior

```
<muxc:TabView Grid.Row="1"
                        x:Name="tabView"
                        VerticalAlignment="Stretch"
                        HorizontalAlignment="Stretch"
                        IsAddTabButtonVisible="False"
                        TabWidthMode="SizeToContent"
                        TabItemsSource="{x:Bind Path=DefaultViewModel.TabItems, Mode=OneWay}"
                        Margin="10">
	<muxc:TabView.TabStripHeader>
		<Button Click="OnAddTabViewItemButtonClick">Add TabViewItem</Button>
	</muxc:TabView.TabStripHeader>
	<muxc:TabView.TabItemTemplate>
		<DataTemplate x:DataType="local:TabItemViewModel">
			<muxc:TabViewItem Header="{x:Bind Header, Mode=OneWay}"
                                        IsClosable="False">
				<Button Margin="5"
                            Content="{x:Bind Header, Mode=OneWay}" />
			</muxc:TabViewItem>
		</DataTemplate>
	</muxc:TabView.TabItemTemplate>
</muxc:TabView>
```

In this case, the `Content` becomes empty when selected tab switches

## Expected behavior

Should not be empty.

## How to reproduce it (as minimally and precisely as possible)

See above

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### LINKED ISSUE #4899: For data-bound TabView, the first item remains selected forever
## Current behavior

When `TabView` items are data-bound via `TabViewItemsSource`, the first item remains in selected state forever, even if switched.

## Expected behavior

Should behave normally.

## How to reproduce it (as minimally and precisely as possible)

Data bind tab view items and select second one.

## Workaround

-

## Environment

Nuget Package:

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

Possibly others too.

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [x] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #4907: TabView items sometimes overlap
## Current behavior

image

## Expected behavior

image

## How to reproduce it (as minimally and precisely as possible)

Samples App -> TabViewPage -> TabView TabItemsSource Page

## Workaround

-

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [x] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): fixes #4895, fixes #4899, fixes #4875, fixes #4907

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Content is not rendered.

## What is the new behavior?

Content is rendered.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs [hunk 1]
```diff
@@ -869,6 +869,13 @@ void OnTabContentPresenterLosingFocus(object sender, LosingFocusEventArgs args)
 
 						tabContentPresenter.LosingFocus += OnTabContentPresenterLosingFocus;
 
+#if IS_UNO
+						// TODO: Uno specific - issue #4894 - in UWP the ContentPresenter does not become
+						// the parent of the Content. In Uno it does, so we need to make sure
+						// the inherited DataContext will match the TabViewItem.
+						tabContentPresenter.DataContext = tvi.DataContext;
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs:876]: Is this possible to test in a runtime test? It'll help when we'll work on ContentControl alignments

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs:876]: You mean to create unit tests that will be disabled for now, testing the correct behavior in general, or to test this workaround?

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs:876]: We need to validate the scenario does not regress, by validating this workaround.

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs:880]: Done 👍

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs [hunk 2]
```diff
@@ -1127,20 +1138,21 @@ void UpdateSelectedItem()
...
-				if (tvi != null)
-				{
-					listView.SelectedItem = tvi;
-
-					// Setting ListView.SelectedItem will not work here in all cases.
-					// The reason why that doesn't work but this does is unknown.
-					tvi.IsSelected = true;
-				}
+				listView.SelectedItem = SelectedItem;
+				//var tvi = SelectedItem as TabViewItem;
```
Comment (davidjohnoliver) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs:1142]: Add a comment indicating why this code is commented out

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**